app/controllers/api/visits_controller.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+module Api
+  class VisitsController < ApplicationController
+    def update
+      visit = Visit.find params[:id]
+
+      validate_happened_at update_params[:happened_at]
```

Comment:
Add in Visit Model `validates :happened_at, format: {with: DateTime.iso8601}, allow_nil: true`